evalml/tests/automl_tests/test_default_algorithm.py

```diff
@@ -385,6 +385,77 @@ def test_select_cat_cols(
             assert value["columns"] == algo._selected_cat_cols
 
 
+@pytest.mark.parametrize(
+    "problem_type,n_unique",
+    [
+        ("binary", 2),
+        ("multiclass", 10),
+        ("multiclass", 200),
+        ("regression", 80),
+        ("regression", 200),
+    ],
+)
+@pytest.mark.parametrize("allow_long_running_models", [True, False])
+def test_default_algorithm_allow_long_running_models_next_batch(
+    allow_long_running_models,
+    problem_type,
+    n_unique,
+    has_minimal_dependencies,
+):
+    if allow_long_running_models and problem_type != "multiclass":
+        pytest.skip("Skipping to shorten tests")
+
+    models_to_check = [
+        "Elastic Net",
+        "XGBoost",
+        "CatBoost",
+    ]
+    if has_minimal_dependencies:
+        models_to_check = ["Elastic Net"]
+
+    X = pd.DataFrame()
+    y = pd.Series([i for i in range(n_unique)] * 5)
+
+    algo = DefaultAlgorithm(
+        X=X,
+        y=y,
+        sampler_name=None,
+        problem_type=problem_type,
+        random_seed=0,
+        allow_long_running_models=allow_long_running_models,
+    )
+    algo._selected_cols = []
+    next_batch = algo.next_batch()
+    found_models = False
+    for pipeline in next_batch:
+        found_models |= any([m in pipeline.name for m in models_to_check])
+
+    # the "best" score will be the 1st dummy pipeline
+    scores = np.arange(0, len(next_batch))
```

Comment:
there's this `add_result` result method on line 84 you can use to clean this up a little.